### PR TITLE
enhance: Refactor channel dist manager interface

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #31091
This PR add GetByFilter interface in channel dist manager, instead of all kind of get func

### CODE REVIEW

File: internal/querycoordv2/checkers/channel_checker.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func (c *ChannelChecker) Check(ctx context.Context) []task.Task {
 		}
 	}
 
-	channels := c.dist.ChannelDistManager.GetAll()
+	channels := c.dist.ChannelDistManager.GetByFilter(nil)
```
Comment (congqixia) [inline internal/querycoordv2/checkers/channel_checker.go:94]: pass not filter is good enough
```suggestion
	channels := c.dist.ChannelDistManager.GetByFilter()
```

File: internal/querycoordv2/meta/channel_dist_manager.go [hunk 2]
```diff
@@ -119,56 +120,28 @@ func (m *ChannelDistManager) GetShardLeadersByReplica(replica *Replica) map[stri
...
 	ret := make([]*DmChannel, 0)
 	for _, channels := range m.channels {
 		for _, channel := range channels {
-			if channel.CollectionID == collectionID {
+			allMatch := true
+			for _, fn := range filters {
+				if fn != nil && !fn(channel) {
+					allMatch = false
+				}
+			}
```
Comment (congqixia) [inline internal/querycoordv2/meta/channel_dist_manager.go:136]: NI: break when some filter returns false
also add a closure here is more readable

**Comment by weiliu1031** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**